src/main/java/hudson/plugins/sauce_ondemand/credentials/SauceCredentials.java

```diff
@@ -111,20 +111,12 @@ public String getDisplayName() {
 
         @SuppressWarnings("unused") // used by stapler
         public FormValidation doCheckApiKey(@QueryParameter String value, @QueryParameter String username) {
-            return doValidate(username, value);
-        }
-
-        @SuppressWarnings("unused") // used by stapler
-        public FormValidation doValidate(@QueryParameter String username, @QueryParameter String apiKey) {
-            if (actualValidate(username, apiKey))
+            SauceREST rest = new JenkinsSauceREST(username, value);
+            if (!rest.getUser().equals("")) {
```

Comment:
Personal style here, I find that ! equals("") confusing to read.

I think with java 7, its safe to compare strings using operators, so do you want to try `rest.getUser() != ""` which might be clearer. Otherwise do you want to make it its own simple function called "isValidCredentials" or something so the conditional is easier to read?